Better Eclipse support

szab100 opened the pull request.

szab100 committed:
Add (better) Eclipse project import support: build.eclipse.xml

szab100 commented:
I re-tested this today and rebased with master changes.
Everything looks good in Eclipse:
  - New project -> Java Project from existing Ant file imports the project with correct classpath, src and test folders.
  - All 1062 tests pass: right click on test folder / Run As -> JUnit Test

wiarlawd commented:
So the purpose of build.eclipse.xml is to create a new Eclipse project. Is that better than have a .project and maybe .classpath file? We did have troubles with those in the past, and eventually deleted them, but just wanted to make sure we'd considered the option.

Could you add in the eventual commit message also a description of using resources instead of local files for the tests. Are these cases where we could be using JUnit 4 temp files instead?

This CL has a lot of 80 column errors, modified whitespace (no-no), and updated import rules (should just be left alone unless we're going to modify all the files in a separate CL, IMHO).

wiarlawd commented:
FYI, OOO for 10 days.

szab100 commented:
Ah okay.. from tomorrow? I'll fix the columns / whitespace issues by the
time you are back..
BTW, what's wrong with those whitespace deletions? Our Eclipse is set up to
auto-fix those, but should just do it at lines we touch..
(the >80 column issue is probably done by that too, since the default
setting is 100, according to google java style guide)

On Wed, Apr 12, 2017 at 5:34 PM, John Lacey <notifications@github.com>
wrote:

> FYI, OOO for 10 days.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/googlegsa/library/pull/16#issuecomment-293744710>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AJicg6xaNP70qwGLsZAv1coq7nnvJyVJks5rvW2IgaJpZM4LNPhK>
> .
>

wiarlawd commented:
It looks like you're deleting trailing whitespace on unmodified lines.

Brett and/or PJ can finish this while I'm out. The library is slated to go out possibly by next Friday.

John L

BrettMichaelJohnson commented on test/com/google/enterprise/adaptor/HeartbeatHandlerTest.java:
```diff
@@ -272,28 +270,32 @@ public void getDocContent(Request request, Response response)
-    HeartbeatHandler handler = createHeartbeatHandlerBuilder()
-        .setWatchdog(watchdog2).setDocHandler(docHandler)
-        .build();
+    mockAdaptor =
```
This is why John is a stickler for arbitrary whitespace changes.  It took me far too long to detect that only the timeouts have changed.

BrettMichaelJohnson reviewed (dismissed):
We are planning to release the library this week, so we would like to wrap up this CL for inclusion.

I don't think John's suggestion re temp files applies here, since the test is not writing to the filesystem. I do agree that an explanation as to switching to locating resources is needed.  I suspect it is because Eclipse isn't setting the current directory when running JUnit.

PapaJoltOmega commented:
Spoke with szab100. Plans to update CL in next couple weeks. There is no need to have this CL in next cut.

PapaJoltOmega commented:
Hi Szabi, can you split of the heart-beat handler test piece.  It's flaky for me right now. Please send as seperate code review for more immediate submission. 

This PR has multiple orthogonal items in it.  Let's submit in peaces please.

szab100 commented:
Yes, sure!

On Tue, May 2, 2017 at 1:34 PM, P.J. O <notifications@github.com> wrote:

> Hi Szabi, can you split of the heart-beat handler test piece. It's flaky
> for me right now. Please send as seperate code review for more immediate
> submission.
>
> This PR has multiple orthogonal items in it. Let's submit in peaces please.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/googlegsa/library/pull/16#issuecomment-298752996>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AJicg3UWX3e1LkRE-Cb9-uVdkt8EjveTks5r15NqgaJpZM4LNPhK>
> .
>

szab100 committed:
Make test resource loading Eclipse compatible

szab100 commented:
@pjogooglecom @wiarlawd @BrettMichaelJohnson 
I just reformatted these changes (eclipse build xml + refactor test resource loading), pls review it again..

PapaJoltOmega commented on test/com/google/enterprise/adaptor/ApplicationTest.java:
```diff
@@ -251,6 +252,15 @@ public void testBasicHttpsListen() throws Exception {
     conn.getContent();
   }
+  
+  private void initSSLKeystores() {
```
is this section simply redundant for non-eclipse users? why don't non-ecloipse users need it?  is there something that can be removed for non-eclipse users? some build.xml lines that are no longer necessary for unit testing?

PapaJoltOmega reviewed (commented):
(1) Flaky test fix was removed from this pull-request.  Can you updatge title of this commit to be only "Better Eclipse Support".  

(2) Can you run this setup without Eclipse to confirm that everythiong works for non-eclipse user and confirm here.

Thank you

szab100 commented:
@pjogooglecom 

1) Done
2) Done - all tests pass with 'ant test' and from Eclipse

Re: initSSLKeyStore() - Yes, this was clearly redundant with the following:
 1) build.xml's "test" target, so I removed it from here (comments retained in TestHelper)
 2) GsaFeedFileSenderTest, so I moved the method to TestHelper and hooked it from both tests

wiarlawd commented:
LGTM

This conflicts with #28, so whoever goes second should pay attention.

Once PJ has approved this, please use "Squash and merge" rather than just clicking the "Merge pull request" button, which has a default of "Create a merge commit".

John L

PapaJoltOmega reviewed (approved):
Thank you
